src/App.tsx

```diff
@@ -1,14 +1,68 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
-import { TodoList } from './components/TodoList';
+import { Todo, TodoStatus, FilterOptions } from './types';
+import { getTodos } from './api';
+import { Loader } from './components/Loader';
 import { TodoFilter } from './components/TodoFilter';
+import { TodoList } from './components/TodoList';
 import { TodoModal } from './components/TodoModal';
-import { Loader } from './components/Loader';
+
+function getFilteredTodos(todos: Todo[], filterOptions: FilterOptions) {
+  const { status, query } = filterOptions;
+  let preparedTodos = [...todos];
+
+  if (status !== TodoStatus.All) {
+    preparedTodos = preparedTodos.filter(({ completed }) => {
+      switch (status) {
+        case TodoStatus.Active:
+          return !completed;
+        case TodoStatus.Completed:
+          return completed;
+        default:
+          throw new Error('This will never happen.');
+      }
+    });
+  }
+
+  if (query) {
+    preparedTodos = preparedTodos.filter(({ title }) => {
+      return title.toLowerCase().includes(query.toLowerCase());
+    });
+  }
+
+  return preparedTodos;
+}
+
+export const initialFilterOptions: FilterOptions = {
+  status: TodoStatus.All,
+  query: '',
+};
 
 export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(true);
```

Comment:
Better to set `false` as default status for loading state, and change it to `true` only before you are doing API call